File: solutions/H.cpp
#include <bits/stdc++.h>
using namespace std;

#define endl '\n'

signed main() {
	ios::sync_with_stdio(0); cin.tie(0);
    int n, m, k, s; cin >> n >> m >> k >> s;
    int ans = 0;
    for (int i = 0; i < m; i++) {
        int x; cin >> x;
        int qtd = 0;
        for (int i = 0; i < n; i++) {
            int bit_x = (x >> i) & 1;
            int bit_s = (s >> i) & 1;
            if (bit_x != bit_s) qtd++;
        }
        if (qtd <= k) ans++;
    }
    cout << ans << endl;
}

File: solutions/H.py
#!/usr/bin/env python3

def main():
    entrada = list(map(int, input().split()))
    n = entrada[0]
    m = entrada[1]
    k = entrada[2]
    s = int(input())
    ans = 0
    for _ in range(m):
        x = int(input())
        qtd = 0
        for i in range(n):
            bit_x = (x >> i) & 1
            bit_s = (s >> i) & 1
            if bit_x != bit_s:
                qtd += 1
        if qtd <= k:
            ans += 1
    print(ans)

if __name__ == '__main__':
    main()